src/uct/ib/rc/accel/rc_mlx5.inl

```diff
@@ -1567,19 +1597,61 @@ uct_rc_mlx5_iface_common_copy_to_dm(uct_rc_mlx5_dm_copy_data_t *cache, size_t hd
     }
 }
 
-static ucs_status_t UCS_F_ALWAYS_INLINE
-uct_rc_mlx5_common_dm_make_data(uct_rc_mlx5_iface_common_t *iface,
-                                uct_rc_mlx5_dm_copy_data_t *cache,
-                                size_t hdr_len, const void *payload,
-                                unsigned length,
-                                uct_rc_iface_send_desc_t **desc_p,
-                                void **buffer_p, uct_ib_log_sge_t *log_sge)
+static void UCS_F_ALWAYS_INLINE uct_rc_mlx5_iface_copy_iov_to_dm(
+        uct_rc_mlx5_dm_copy_data_t *cache, size_t hdr_len, const uct_iov_t *iov,
+        size_t iovcnt, void *dm, uct_ib_log_sge_t *log_sge)
+{
+    uint64_t word = 0;
+    size_t offset;
+
+    ucs_assert(sizeof(cache->bytes) >= hdr_len);
+
+    /* copy head of iov to tail of cache */
+    offset = uct_iov_to_buffer(iov, iovcnt, 0,
+                               UCS_PTR_BYTE_OFFSET(cache->bytes, hdr_len),
+                               sizeof(cache->bytes) - hdr_len);
+
+    UCS_STATIC_ASSERT(sizeof(*cache) == sizeof(cache->bytes));
+    UCS_STATIC_ASSERT(sizeof(log_sge->sg_list) /
+                              sizeof(log_sge->sg_list[0]) >= 2);
+
+    /* atomically by 8 bytes copy data to DM */
+    /* cache buffer must be aligned, so, source data type is aligned */
+    UCS_WORD_COPY(volatile uint64_t, dm, uint64_t, cache->bytes,
+                  sizeof(cache->bytes));
+    dm += sizeof(cache->bytes);
+    if (ucs_log_is_enabled(UCS_LOG_LEVEL_TRACE_DATA)) {
+        log_sge->sg_list[0].addr   = (uint64_t)cache;
+        log_sge->sg_list[0].length = (uint32_t)hdr_len;
+
+        log_sge->sg_list[1].addr   = (uint64_t)iov->buffer;
+        log_sge->sg_list[1].length = (uint32_t)iov->length;
+
+        log_sge->num_sge = 2;
+    } else {
+        log_sge->num_sge = 0;
+    }
+
+    while (uct_iov_to_buffer(iov, iovcnt, offset, &word, sizeof(word))) {
```

Comment:
`while (uct_iov_to_buffer(iov, iovcnt, offset, &word, sizeof(word)) != 0) {`
we use direct comparison with `0` for non-boolean (by boolean I mean logically boolean value)